File: site/ng2-app/app/room/room-info.ts
export class RoomInfo {

    public status: string = 'Closed';

    public key: string = null;

    public owner: string = null;

    constructor(public id: number,
                public name: String) {

    }

    public static fromJson(json: any): RoomInfo {
        let room = new RoomInfo(json["id"], json["name"]);
        room.owner = json.owner.userName;
        room.key = json.key;
        if ("status" in json) {
            room.status = (<string>json["status"]);
        }
        return room;
    }
}

File: KwizzApi/Models/RoomInfo.cs
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace KwizzApi.Models
{
    public class RoomInfo
    {
        public long Id { get; set; } = 0;

        [Required]
        public string Name { get; set; }

        public string Key { get; set; }

        [JsonConverter(typeof(StringEnumConverter))]
        public RoomStatus Status { get; set; } = RoomStatus.Closed;

        public ApplicationUser Owner { get; set; }
    }
}